### PR TITLE
feat: Add a command to generate JSON Schema

### PR DESCRIPTION
**Pull request by haoel**
The JSON Schema could help to edit the EaseProbe's configuration YAML file easier - we can use it to generate the UI form on the Web or just edit the YAML file via IDE like VSCode or Intellij IDEA.

This PR just generates the EaseProbe YAML Configuration JSON Schema

- Support `-j` flag to generate the EaseProbe JSON Schema file.
- The generated scheme file is uploaded to `resources/schema.json`

### CODE REVIEW

File: conf/conf.go [hunk 1]
```diff
@@ -86,60 +88,90 @@ func (s *Schedule) UnmarshalYAML(unmarshal func(interface{}) error) error {
 
 // Notify is the settings of notification
 type Notify struct {
-	Retry global.Retry `yaml:"retry"`
-	Dry   bool         `yaml:"dry"`
+	Retry global.Retry `yaml:"retry" json:"retry,omitempty" jsonschema:"title=retry,description=the retry settings"`
+	Dry   bool         `yaml:"dry" json:"dry,omitempty" jsonschema:"title=dry,description=if true, the notification will not be sent,default=false"`
```
Comment (localvar) [inline conf/conf.go:92]: there's a comma in the description, while jsonschema uses commas as delimiters, this results in a wrong description in the generated JSON schema (refer L181 of schema.json).
and there are other cases.

File: notify/log/log.go [hunk 2]
```diff
@@ -47,11 +47,11 @@ const (
...
 	base.DefaultNotify `yaml:",inline"`
 
-	File    string `yaml:"file"`
-	Host    string `yaml:"host"`
-	Network string `yaml:"network"`
-	Type    Type   `yaml:"-"`
-	logger  *log.Logger
+	File    string      `yaml:"file" json:"file,omitempty" jsonschema:"title=Log File,description=The log file to write the notification message"`
+	Host    string      `yaml:"host" json:"host,omitempty" jsonschema:"title=Syslog Host,description=The log host to write the notification message"`
+	Network string      `yaml:"network" json:"network,omitempty" jsonschema:"title=enum=tcp,enum=udp,Syslog Network,description=The log network to write the notification message"`
```
[the remark this answers is not in the document]
Comment (haoel) [reply notify/log/log.go]: very good catch

File: resources/schema.json [hunk 3]
```diff
@@ -0,0 +1,1990 @@
...
+          "description": "AWS Credential"
+        },
+        "profile": {
+          "type": "string",
+          "title": "AWS Profile",
+          "description": "AWS Profile"
+        },
+        "format": {
+          "type": "integer",
+          "enum": [
```
Comment (zhao-kun) [inline resources/schema.json:589]: All values of the enum both are `0`? It looks something was wrong.

Comment (haoel) [reply resources/schema.json:589]: thanks, I will fix this

File: docs/Manual.md [hunk 4]
```diff
@@ -1329,4 +1331,27 @@ settings:
...
   # https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
   timezone: "America/New_York" #  default: UTC
-```
\ No newline at end of file
+```
+# 8. Tools
+
+## 8.1 EaseProbe JSON Schema
+
+We have a schema file can be used to validate the EaseProbe configuration file. It is available at [resources/schema.json](https://raw.githubusercontent.com/megaease/easeprobe/main/resources/schema.json).
```
Comment (proditis) [inline docs/Manual.md:1339]:
```suggestion
We have a JSON schema that can be used to validate your EaseProbe configuration. The schema can be found at [resources/schema.json](https://raw.githubusercontent.com/megaease/easeprobe/main/resources/schema.json).
```

File: docs/Manual.md [hunk 5]
```diff
@@ -1329,4 +1331,27 @@ settings:
...
+
+This schema file can be generated by the following command:
```
Comment (proditis) [inline docs/Manual.md:1341]:
```suggestion
The schema file can be generated at any time running the following command:
```

File: docs/Manual.md [hunk 6]
```diff
@@ -1329,4 +1331,27 @@ settings:
...
+
+```bash
+$ easeprobe -j > schema.json
+```
+
+For VSCode, you can install the [YAML extension](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml) and add the following configuration to your `settings.json` file:
```
Comment (proditis) [inline docs/Manual.md:1347]:
```suggestion
In order to use the schema with VSCode for validating your configuration, you need to install the [YAML extension](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml) and add the following configuration to your `settings.json` file:
```

**Review by proditis — APPROVED** [review]
LGTM

File: docs/Manual.md [hunk 7]
```diff
@@ -1336,15 +1336,15 @@ settings:
 
 ## 8.1 EaseProbe JSON Schema
 
-We have a schema file can be used to validate the EaseProbe configuration file. It is available at [resources/schema.json](https://raw.githubusercontent.com/megaease/easeprobe/main/resources/schema.json).
+We have a JSON schema that can be used to validate your EaseProbe configuration. The schema can be found at [resources/schema.json](https://raw.githubusercontent.com/megaease/easeprobe/main/resources/schema.json).
 
-This schema file can be generated by the following command:
+The schema file can be generated at any time running the following command:
```
Comment (proditis) [inline docs/Manual.md:1341]:
```suggestion
The schema file can be generated at any time by running the following command:
```

**Review by zhao-kun — APPROVED** [review]

**Pull request merged**